src/events/handlers/CountingHandler.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  Events,
+  type ClientEvents,
+  Message,
+  type OmitPartialGroupDMChannel,
+  type EmojiIdentifierResolvable, Emoji, GuildEmoji, ReactionEmoji, type User, type Interaction
+} from "discord.js";
+import {Logger, type ILogObj} from "tslog";
+import type IEvent from "../IEvent.ts";
+import type SnugglesClient from "../../structure/Client.ts";
+import GuildCountingService from "../../database/services/GuildCountingService.ts";
+import {column, count, evaluate, number, re} from 'mathjs'
+import type {GuildCounting} from "@prisma/client";
+
+
+export const logger: Logger<ILogObj> = new Logger();
+
+export class CountingHandler implements IEvent {
+  public event: keyof ClientEvents = Events.MessageCreate;
+  public once = false;
+
+  constructor(private client: SnugglesClient) {}
+
+  async handle(message: OmitPartialGroupDMChannel<Message<boolean>>) {
+    const countData = await GuildCountingService.getCountingDataForChannel(message.channelId);
+    if (!countData) return;
+    logger.debug(`message: ${message.content} | count: ${countData.count}`)
+
+    // Ignore bot messages and webhooks
+    if (message.webhookId != null) return
+    if (message.author.bot) return
+
+    let number;
+
+    try {
+      number = evaluate(message.content)
+    } catch (error) {
+      number = NaN
+    }
+
+    if (isNaN(number)) {
+      // Unable to evaluate - treat as regular chat message
+      if (!countData.allowTalkingUsers) await message.delete();
+      return;
+    }
+
+    const lastCounter = countData.lastCounterId
+    countData.lastCounterId = message.author.id
+
+    const isCorrect = number === countData.count + 1
+    let currentCount = countData.count
+
+    if (!isCorrect) {
+      countData.count = 0
+      await GuildCountingService.updateCountData(countData)
+      await message.reply(this.parseCountingMessage(countData.wrongNumberFailMessage, message.author, currentCount))
+      await message.react(this.getReaction(countData, isCorrect))
+      return
+    }
+
+    if (currentCount != 0 && !countData.allowConsecutiveCounts && lastCounter === message.author.id) {
+      countData.count = 0
+      await GuildCountingService.updateCountData(countData)
+      await message.reply(this.parseCountingMessage(countData.consecutiveCountingFailMessage, message.author, currentCount))
+      await message.react(this.getReaction(countData, isCorrect))
+      return
+    }
```

Comment:
the above two if-statements have very similar functions...

Reply:
i made it _"expandable"_ ✨✨

Reply:
it doesn't look very expandable

Reply:
well it is